beacon_node/beacon_chain/src/beacon_block_reward.rs

```diff
@@ -0,0 +1,258 @@
+use crate::{BeaconChain, BeaconChainError, BeaconChainTypes};
+use operation_pool::{earliest_attestation_validators, SplitAttestation};
+use safe_arith::SafeArith;
+use slog::error;
+use state_processing::{
+    common::{
+        altair, base, get_attestation_participation_flag_indices, get_attesting_indices,
+        get_attesting_indices_from_state,
+    },
+    per_block_processing::{
+        altair::sync_committee::compute_sync_aggregate_rewards, get_slashable_indices,
+    },
+};
+use store::consts::altair::{PARTICIPATION_FLAG_WEIGHTS, PROPOSER_WEIGHT, WEIGHT_DENOMINATOR};
+use types::{
+    beacon_state::BeaconStateBase, BeaconBlockRef, BeaconState, BeaconStateError, ExecPayload,
+};
+
+type BeaconBlockSubRewardValue = u64;
+
+impl<T: BeaconChainTypes> BeaconChain<T> {
+    pub fn compute_beacon_block_reward<Payload: ExecPayload<T::EthSpec>>(
+        &self,
+        block: BeaconBlockRef<'_, T::EthSpec, Payload>,
+        state: &mut BeaconState<T::EthSpec>,
+    ) -> Result<
+        (
+            BeaconBlockSubRewardValue,
+            BeaconBlockSubRewardValue,
+            BeaconBlockSubRewardValue,
+            BeaconBlockSubRewardValue,
+            BeaconBlockSubRewardValue,
+        ),
+        BeaconChainError,
+    > {
+        if block.slot() != state.slot() {
+            return Err(BeaconChainError::BlockRewardSlotError);
+        }
+
+        let sync_aggregate_reward =
+            self.compute_beacon_block_sync_aggregate_reward(block, state)?;
+
+        let proposer_slashing_reward = self
+            .compute_beacon_block_proposer_slashing_reward(block, state)
+            .map_err(|e| {
+                error!(
+                self.log, "Error calculating proposer slashing reward";
+                "error" => ?e
+                );
+                BeaconChainError::BlockRewardSyncError
+            })?;
+
+        let attester_slashing_reward = self
+            .compute_beacon_block_attester_slashing_reward(block, state)
+            .map_err(|e| {
+                error!(
+                self.log, "Error calculating attester slashing reward";
+                "error" => ?e
+                );
+                BeaconChainError::BlockRewardSyncError
```

Comment:
The "Sync" in this error is unnecessary 

```suggestion
                BeaconChainError::BlockRewardError
```